ray-operator/controllers/ray/common/pod.go

```diff
@@ -51,17 +51,10 @@ func GetHeadPort(headStartParams map[string]string) string {
 	return headPort
 }
 
-// rayClusterHAEnabled check if RayCluster enabled FT in annotations
-func rayClusterHAEnabled(instance rayv1alpha1.RayCluster) bool {
-	if instance.Annotations == nil {
```

Comment:
The nil map in Golang can be a bit tricky. It won't throw an exception if you look up a nil map, but it will throw an exception when you attempt to write to a nil map.

> A nil map behaves like an empty map when reading, but attempts to write to a nil map will cause a runtime panic;
* ref: https://go.dev/blog/maps


* example:
   ```go
  package main
  
  import "fmt"
  
  func main() {
	  var rect map[string]int
	  val, ok := rect["hi"] 
	  fmt.Println(val, ok) // 0, false
	  rect["height"] = 10 // panic: assignment to entry in nil map
	  fmt.Println(rect["height"])
  }
   ```